lang/test/org/partiql/lang/eval/EvaluatorTestSuite.kt

```diff
@@ -1398,4 +1398,38 @@ internal val EVALUATOR_TEST_SUITE: IonResultTestSuite = defineTestSuite {
             "$partiql_bag::[[1968, 4, 3, 12, 31, 59]]"
         )
     }
+    group("arithmetic with mixed type") {
```

Comment:
Looking at the other uses of `resolveIntConstraint`, could add tests cases for unary plus, unary minus, modulo, and function calls.